docs/utilities/parser.md

```diff
@@ -169,6 +155,14 @@ def my_function():
 		}
 ```
 
+#### Arbitrary data model parsing
+
+The parser allows you parse events into arbitrary data structures, such as `dict` or classes that don't inherit from BaseModel. The following example shows you how to parse a [`Union`](https://docs.pydantic.dev/latest/api/standard_library_types/#union):
```

Comment:
I think we can do better here and add a link to the TypeAdapter documentation.

Reply:
I don't want to add the TypeAdapter ref, seems like an implementation detail – plus their docs don't really give any useful info